Fixing brainmodel and accommodating non-standard cifti

mychan24 opened the pull request:
**Issue (1)**. Using read_cifti() on a cifti file with both surface and volume based structures only reads in the attributes of the surface structures. 
- **Edited parse_brain_model.R**  

**Issue (2)**. dtseries.nii from MSC (Midnight Scanning Club) has non-standard dimension where the dense data is on the first dimension, so when the data is assigned into an array, it is arranged incorrectly. 

- Edited **cifti_data.R** with comments documenting why. This does not affect cifti with standard dimensions. Two .md vignettes were added to show the error using the original cifti functions and the updated functions. The vignettes also show that the updated script does not affect how standard cifti are loaded in (e.g., data from download_cifti_data)

Given people may use these function with the MSC data, I think its worth incorporating it into the function.

muschellij2 commented:
Thanks for the PR.  Before accepting please formats the vignettes to Rmd files (see `usethis::use_vignette` for the header).  Also, for my packages, I need them to be fully reproducible (even the vignettes).  Therefore, we need to include the cifti in the package or have them uploaded somewhere like figshare.

mychan24 commented:
Edited vignettes to be .Rmd and with correct header. It will also download the test data from its source from openneuro.org if it is not available locally. I checked that it is reproducible on separate machines. This is my first time trying to use vignettes correctly, so if it's still not in the correct format, please let me know. Thanks.

mychan24 commented:
Hi, I added the Rmd and moved them into a folder call test_Rmd instead of letting them stay in the vignette. As they are not really how-tos or vignettes meant to distribute to CRAN. It was more to document the issue for this particular update, and if it works well, we could eventually remove them. 
The Rmd will download the needed data from openneuro if it is not locally available, and is reproducible now on different machines.
